plugins/outputs/influxdb_v2/errors.go

```diff
@@ -0,0 +1,51 @@
+package influxdb_v2
+
+import (
+	"fmt"
+	"time"
+)
+
+type genericRespError struct {
+	Code      string
+	Message   string
+	Line      *int32
+	MaxLength *int32
+}
+
+func (g genericRespError) Error() string {
+	errString := fmt.Sprintf("%s: %s", g.Code, g.Message)
+	if g.Line != nil {
+		return fmt.Sprintf("%s - line[%d]", errString, g.Line)
+	} else if g.MaxLength != nil {
+		return fmt.Sprintf("%s - maxlen[%d]", errString, g.MaxLength)
+	}
+	return errString
+}
+
+type APIError struct {
+	Err        error
+	StatusCode int
+	Retryable  bool
+}
+
+func (e APIError) Error() string {
+	return e.Err.Error()
+}
```

Comment:
Potential panic if e.Err is nil
```suggestion
func (e APIError) Error() string {
    if e.Err == nil {
        return fmt.Sprintf("API error: status %d", e.StatusCode)
    }
    return e.Err.Error()
}
```